tiny-firmware/firmware/fsm_impl.c

```diff
@@ -213,7 +215,9 @@ ErrCode_t msgSignTransactionMessageImpl(uint8_t *message_digest, uint32_t index,
     uint8_t pubkey[SKYCOIN_PUBKEY_LEN] = {0};
     uint8_t seckey[SKYCOIN_SECKEY_LEN] = {0};
     uint8_t signature[SKYCOIN_SIG_LEN];
-    ErrCode_t res = fsm_getKeyPairAtIndex(1, pubkey, seckey, NULL, index);
+    int (*address_from_pubkey)(const uint8_t*, char*, size_t*);
+    address_from_pubkey = &skycoin_address_from_pubkey;
+    ErrCode_t res = fsm_getKeyPairAtIndex(1, pubkey, seckey, NULL, index, address_from_pubkey);
```

Comment:
You can omit creating `address_from_pubkey` variable and directly path `&skycoin_address_from_pubkey` to fsm_getKeyPairAtIndex